apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,8 +278,25 @@ type BareMetalHostSpec struct {
 	// the power status and hardware inventory inspection. If the
 	// Image field is filled in, this field is ignored.
 	ExternallyProvisioned bool `json:"externallyProvisioned,omitempty"`
+
+	// When set to Disabled, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=Full
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
```

Comment:
you could make this a pointer to simplify figuring out if it is omitted

Reply:
Added a check for nil pointer rather as you suggested. Thanks.